### DISCUSSION TITLE
Handlebar planner seems difficult to use overall, comparing with functional calling with auto invoke or stepwise planner.

### DISCUSSION THREAD

**Discussion by sbkkalex** [microsoft/semantic-kernel#5672] [General]
Any good reason to use a handlebar planner?

What I found is that it is hard for handlebar planner to work as expect due to the reason:
- Need very specific plugins. For example, need a json parser plugin to properly with api. https://github.com/microsoft/semantic-kernel/issues/5669
- Because the plan is generated in one call, when things not going well, it cannot adjust it's strategy. For example, usually return a pure json string.
- Buggy (Temple helper function hallucination, type error, ext...)

On the other hand, functional calling with auto invoke and stepwise planner just works.

**Comment by sophialagerkranspandey** [comment]
The difference between the two is the Handlebars planner allows you to inspect and approve a plan from the LLM before it's invoked. As you noted, sometimes the LLM generates improper plans. We are currently investigating how to leverage the strategies the assistant API uses with code interpreter to improve our planners.

**Comment by sbkkalex** [comment]
@sophialagerkranspandey 
Thank you for your response. Good to hear future improment plans.👍
